### DISCUSSION TITLE
Prevent object(s) for being zoomed

### DISCUSSION THREAD

**Discussion by KhorenGrig** [fabricjs/fabric.js#7584] [Q&A] [answered]
Is there any native way to prevent specific object(s) to be zoomed, when using these methods (`zoomToPoint, setZoom`), Tried [https://stackoverflow.com/questions/48578974/maintaining-object-size-while-zooming-in-fabric-js](url) and [https://pretagteam.com/question/maintaining-object-size-while-zooming-in-fabric-js](url) suggestions, but did not work as I wanted for my case.

The problem is I have three objects(left ruler, top ruler, and overlayImage). When I do zoom I need to clear these objects and redraw again(based on canvas scale), but these objects are stretching and changing their positions after redraw (Which I do not want)

**Comment by asturur** [comment, marked as answer]
there are 2 ways:
- put ruler, top ruler and overlayImage in the overlayImage, and disable vpt for the overlays ( there is a StaticCanvas option )
- override the render overlay method on the static canvas and do what you need to do there manually ( render a ruler and other things that may be fabricObjects or lower level drawing calls )

**Reply by KhorenGrig** [reply]
@asturur Thanks a lot for your reply. I will try your suggestions.

One more question is also.

Why conditionally rendering canvas **on DOM** brings App crush(for my case it's ReactJS application). Mounting works well, but unmounting don't.  If you are familiar a little bit with react pls see the sandbox (https://codesandbox.io/s/clever-hermann-klckc?file=/src/App.js), otherwise there is no problem, you can skip my question.

**Reply by asturur** [reply]
Not sure what are you doing with that use Effect.
https://github.com/asturur/qr-roguer/blob/master/src/FabricCanvas.jsx
Have a look at this repo, is a project i started and i m not working on it, but there is an example on how to use react with fabric and does not crash.

**Reply by KhorenGrig** [reply]
Wrapping `canvas` into a single `div` solves a problem (did not figure out why) , Like this  `canvas id="canvas" />` -> `<div><canvas id="canvas" /></div>`.  _Again thanks a lot_ . 

P.S. I decided to inject another canvas for ruler and grid (in order to escape from above  zoom problems), and that component(which renders canvas), should be conditionally rendered (via checkbox true/false).

**Comment by umtdemr** [comment]
@asturur Is there any solution for the other objects that we want to prevent being zoomed? For example preventing a rectangle being zoomed. 

Maybe changing object scaleX and scaleY to ```1 / canvas.getZoom()```  can solve the problem but that solution is changing position of the object while changing the scale properties.

**Reply by puschie286** [reply]
we use this approach with some shadow properties and custom scaling function to ensure these properties are in sync.
